pkg/services/updatechecker/instrumentation/httpclient.go

```diff
@@ -0,0 +1,128 @@
+package instrumentation
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel/codes"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/grafana/grafana/pkg/infra/tracing"
+)
+
+// InstrumentedHTTPClient is an HTTP client that wraps every request with a span, tracking the method, url,
+// and response code as span attributes.
+type InstrumentedHTTPClient struct {
+	*http.Client
+	tracer tracing.Tracer
+
+	// spanName is the name that request spans will have.
+	spanName string
+
+	metrics *PrometheusMetrics
+}
+
+// defaultSpanName is the default span name for spans created by InstrumentedHTTPClient
+const defaultSpanName = "InstrumentedHTTPClient.request"
+
+// NewInstrumentedHTTPClient returns a new usable InstrumentedHTTPClient
+func NewInstrumentedHTTPClient(cl *http.Client, tracer tracing.Tracer, opts ...InstrumentedHTTPClientOption) *InstrumentedHTTPClient {
+	r := &InstrumentedHTTPClient{
+		Client:   cl,
+		tracer:   tracer,
+		spanName: defaultSpanName,
+	}
+	for _, opt := range opts {
+		opt(r)
+	}
+	return r
+}
+
+// InstrumentedHTTPClientOption is a function that mutates the provided InstrumentedHTTPClient
+type InstrumentedHTTPClientOption func(cl *InstrumentedHTTPClient)
+
+// WithSingleMetrics takes some prometheus collectors and sets them into cl.metrics
+func WithSingleMetrics(
+	requestsCounter, failureCounter prometheus.Counter,
+	durationSecondsHistogram prometheus.Histogram,
+	inFlightGauge prometheus.Gauge,
+) InstrumentedHTTPClientOption {
+	return func(cl *InstrumentedHTTPClient) {
+		cl.metrics = &PrometheusMetrics{
```

Comment:
Maybe personal opinion. Given we have logs capturing this don't think we need metrics for it. Might simplify the code a bit as well.

WDYT?

Reply:
Removed as suggested because we can create those metrics from loki as well :+1: